SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -425,32 +426,35 @@ - (void)sendMessage:(TSOutgoingMessage *)message
             success:(void (^)())successHandler
             failure:(void (^)(NSError *error))failureHandler
 {
-    [self sendMessage:message transaction:nil success:successHandler failure:failureHandler];
-}
-
-- (void)sendMessage:(TSOutgoingMessage *)message
-        transaction:(YapDatabaseReadWriteTransaction *_Nullable)transaction
-            success:(void (^)())successHandler
-            failure:(void (^)(NSError *error))failureHandler
-{
     OWSAssert(message);
 
-    if (transaction) {
-        [message updateWithMessageState:TSOutgoingMessageStateAttemptingOut transaction:transaction];
-    } else {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
There's a few steps to sending a message - try to do as many of them as possible off the main thread.  They need to be performed serially, so we get a bit of "callback hell".